### ISSUE TITLE
Piping Stream to Response Yields Empty Result

### ISSUE THREAD

**Issue by JD-Robbs** [pleerock/routing-controllers#24]
Consider the following example of piping a stream (any stream; but my example uses PDF) to the result object:

```
@Get("/pdf")
public pdf(@Req() req: Request, @Res() res: Response) {
    let wkhtmltopdf = require("wkhtmltopdf");

    res.setHeader("Content-type", "application/pdf");
    wkhtmltopdf("http://localhost/my/app").pipe(res);
}
```

Unfortunately, the route simply returns `Cannot GET /pdf`.

The same code, however, works when conventionally registering the route as below. 
I wonder if I'm doing something wrong, or if the piping of streams is not yet supported.

```
app.get("/test", (req: Request, res: Response) => {
    let wkhtmltopdf = require("wkhtmltopdf");

    res.setHeader("Content-type", "application/pdf");
    wkhtmltopdf("http://localhost/projects/proposely").pipe(res);
});
```

**Comment by pleerock** [comment]
I haven't tested pipes support, I need deeper investigation of this issue. Do you know what `wkhtmltopdf("...").pipe(res);` does inside? Does it use request.send somehow?

btw this is offtopic, but you can use `@Header("Content-type", "application/pdf")` instead of `res.setHeader("Content-type", "application/pdf");`

**Comment by Diluka** [comment]
If I call `res.send` manually, there must be an error `Can't set headers after they are sent.`

**Comment by felipesabino** [comment]
Any news on this? I am having the same issue trying to return a stream with binary data

```
Error: write after end
```

**Comment by pleerock** [comment]
I recommend to use old school `res` and return using it until I come back to this issue.

**Comment by ajmalmsali** [comment]
@felipesabino  any updates?

**Comment by pleerock** [comment]
this should be fixed in 0.7.0. If its not please create a new issue.
